src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java

```diff
@@ -71,12 +99,171 @@ public void validateLocation(final BindingResult errors, final Integer locationI
 			throw new ApiRequestValidationException(errors.getAllErrors());
 		}
 
-		final Location location = locationDataManager.getLocationByID(locationId);
+		final Location location = this.locationDataManager.getLocationByID(locationId);
 
 		if (location == null) {
 			errors.reject("location.invalid", "");
 			throw new ApiRequestValidationException(errors.getAllErrors());
 		}
 	}
 
+	public void validateCreation(final LocationRequestDto locationRequestDto) {
+		this.errors = new MapBindingResult(new HashMap<>(), LocationRequestDto.class.getName());
+
+		this.validateLocationType(locationRequestDto.getType());
+		this.validateLocationAbbr(locationRequestDto.getAbbreviation());
+
+		if (locationRequestDto.getProvinceId() != null) {
```

Comment:
@cuenyad 
create a private function named validateLocationId and send the property as a parameter, this can be reused in 4 places in this new code. 
It can also be reused in `validateLocation`